src/k8s/pkg/k8sd/api/cluster_tokens.go

```diff
@@ -40,6 +40,26 @@ func postClusterJoinTokens(m *microcluster.MicroCluster, s *state.State, r *http
 	return response.SyncResponse(true, &apiv1.TokensResponse{EncodedToken: token})
 }
 
+func getOrCreateJoinTocken(m *microcluster.MicroCluster, tokenName string) (string, error) {
+	// grab token if it exists and return it
+	records, err := m.ListJoinTokens()
+	if err != nil {
+		return "", err
+	}
+	for _, record := range records {
+		if record.Name == tokenName {
+			return record.Token, nil
+		}
+	}
+
+	// if token does not exist, create a new one
+	token, err := m.NewJoinToken(tokenName)
+	if err != nil {
+		return "", err
```

Comment:
wrap the error and add relevant context, e.g.
```suggestion
		return "", fmt.Errorf("failed to generate a new microcluster join token: %w", err)
```